pkg/pillar/types/volumetypes.go

```diff
@@ -182,3 +188,59 @@ func (status VolumeStatus) LogDelete() {
 func (status VolumeStatus) LogKey() string {
 	return string(base.VolumeStatusLogType) + "-" + status.Key()
 }
+
+// VolumeRefConfig : Reference to a Volume specified separately in the API
+// If a volume is purged (re-created from scratch) it will either have a new
+// UUID or a new generationCount
+type VolumeRefConfig struct {
+	VolumeID          uuid.UUID
+	GenerationCounter int64
+	RefCount          uint
+}
+
+// Key : VolumeRefConfig unique key
+func (config VolumeRefConfig) Key() string {
+	return fmt.Sprintf("%v#%d", config.VolumeID, config.GenerationCounter)
```

Comment:
Shouldn't this be %s to print the UUID as a string? %v might print as a sequence of bytes for all I know.

Reply:
Resolved.